fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/DistributionPruneRule.java

```diff
@@ -52,7 +53,8 @@ public DistributionPruneRule() {
         }
 
         // prune hint tablet
-        if (null != olapScanOperator.getHintsTabletIds() && !olapScanOperator.getHintsTabletIds().isEmpty()) {
+        Preconditions.checkState(olapScanOperator.getHintsTabletIds() != null);
```

Comment:
Why check hint tablet？It's maybe null

Reply:
Yes.  I add code here

```
 private List<Long> tabletIds = Lists.newArrayList();
```

Reply:
But the code in `TableRef`, not `LogicalOlapScanOperator`,  construct method of `LogicalOlapScanOperator` can't sure `hintsTabletIds` always not null, and I find many call without set `hintsTabletIds` in UT. Maybe we need add default value in `LogicalOlapScanOperator` or add @TODO to deal it later ?